### PR TITLE
Adds getting started in Readme.MD

### PR DESCRIPTION
**Pull request by mustafaiman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+        - `<multicast>`: Members find each other by sending multicast requests to the specified address and port. It is very useful if ip addresses
+        of the members are not static.
+        - `<tcp>`: This strategy uses a pre-configured list of known members to find and already existing cluster. It is enough for a member to
+        find only one cluster member to connect to the cluster. The rest of the member list is automatically retrieved from that member. We recommend
+        putting multiple known member addresses there to avoid disconnectivity should one of the members in the list is unavailable at the time
+        of connection.
+
+These configuration elements are enough for most connection scenarios. Now we will move onto configuration of the Node.js client.
+
+## Hazelcast Client Configuration
```
Comment (furkansenharputlu) [inline README.md]: This part is similar with configuration options in the PR #342 .

Comment (mustafaiman) [reply README.md]: This part looks like repeated in more than two places. I think redundancy in a guide is not a bad thing though. I say let us keep redundant parts for now. If we find them distracting or completely useless, we can remove them at the end.

File: README.md [hunk 2]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+and the following sections for information about detailed network configuration and/or additional features of Hazelcast Node.js Client
+configuration.
+
+An easy way to configure your Hazelcast Node.js Client is to create a `Config` object and set the appropriate options. Then you can
+supply this object to your client at the startup. Another way to configure your client is to provide a `hazelcast-client.json` file. This approach is similar to `hazelcast.xml` approach
+in configuring the member. Note that `hazelcast-client.json` is a JSON file whereas member configuration is XML based. Although these
+two formats are different, you will realize that the names of the configuration parameters are the same for both client and the member.
+It is done this way to make it easier to transfer Hazelcast skills to multiple platforms easily.
+
+Once you embedded hazelcast-client to your Node.js project. You may follow any of programmatic or declarative configuration approaches.
```
Comment (Serdaro) [inline README.md]: Once you embedded hazelcast-client to your Node.js project, you may follow...

File: README.md [hunk 3]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+with the location of your config file. In this case, the client uses the configuration file specified in the environment variable.
+
+For the structure of `hazelcast-client.json`, take a look at [hazelcast-client-full.json](test/config/hazelcast-client-full.json). You
+can use only the relevant parts of the file in your `hazelcast-client.json` and remove the rest. Default configuration is used for any
+part that you do not explicitly set in `hazelcast-client.json`.
+
+---
+
+If you run Hazelcast IMDG members in a different server than the client, you most probably have configured the members' ports and cluster
+names as explained in the previous section. If you did, then you need to make certain changes to network settings of your client.
```
Comment (Serdaro) [inline README.md]: ...to the network settings of...

File: README.md [hunk 4]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+        "clusterMembers": [
+            "some-ip-address:port"
+        ],
+    }
+}
 
-Hazelcast Node.js Client connects to a Hazelcast IMDG cluster. See [https://hazelcast.org/download/](https://hazelcast.org/download/).
+## Basic Usage
+Now that we have a working cluster and we know how to configure both our cluster and client, we can run a simple program to use a
+distributed map in Node.js
```
Comment (Serdaro) [inline README.md]: ...in Node.js client.

File: README.md [hunk 5]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+# Basic Configuration
+If you are using Hazelcast IMDG and Node.js Client on the same computer, generally default configuration just works. This is great for
+trying out the client and experimenting. However, if you run the client on a different computer than any of the cluster members, you may
+need to do some simple configuration such as member addresses.
+
+IMDG members and the clients has their own configuration options. You may need to reflect some configuration options that you made on member
+side to client side in order to connect to the cluster. This section describes the most common configuration elements to get you started in no time.
+It discusses some member side configuration options to ease understanding Hazelcast's ecosystem. Then, client side configuration options
+regarding cluster connection are discussed. Configuration material regarding data structures are discussed in the following sections.
+You can refer to [IMDG Documentation](https://hazelcast.org/documentation/) and [CONFIG.md](CONFIG.MD) for more information.
```
[the remark this answers is not in the document]
Comment (mustafaiman) [reply README.md]: I do not think `CONFIG.md`s whole content is going to be moved here. `CONFIG.md` discusses all details of configuration, not only common cases.

File: README.md [hunk 6]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+
+## IMDG Configuration
+There are two ways to configure Hazelcast IMDG. One is to use a `hazelcast.xml` file and the other is to programmatically configure the
+instance before starting it from Java code. Since we use standalone servers, we will use `hazelcast.xml` to configure our cluster members.
```
[the remark this answers is not in the document]
Comment (mustafaiman) [reply README.md]: I am adding a new parahgraph just above this one about why we may want to configure standalone servers

File: README.md [hunk 7]
```diff
@@ -59,54 +59,312 @@ Hazelcast Node.js client supports the following data structures and features:
...
+
+When you download and unzip hazelcast-<version>.zip, you see the `hazelcast.xml` in /bin folder. When a Hazelcast member starts, it looks for
+`hazelcast.xml` file to load configuration from. In `hazelcast.xml` file, you will see the default configuration. We will go over some important
```
[the remark this answers is not in the document]
Comment (mustafaiman) [reply README.md]: absolutely good suggestion

File: README.md [hunk 8]
```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
...
+part that you do not explicitly set in `hazelcast-client.json`.
+
+---
+
+If you run Hazelcast IMDG members in a different server than the client, you most probably have configured the members' ports and cluster
+names as explained in the previous section. If you did, then you need to make certain changes to the network settings of your client.
+
+### Group Settings
+
+- Programmatic
```
Comment (furkansenharputlu) [inline README.md]: Using this style is better to be compatible with other PRs.
**Programmatic:**
```javascript
let cfg = new Config.ClientConfig();
cfg.group.name = // group name of you cluster
```

**Declarative:**
```json
{
    "group": {
        "name": "group name of you cluster"
    }
}
```

File: README.md [hunk 9]
```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
...
+```
```
Comment (furkansenharputlu) [inline README.md]: `javascript`

File: README.md [hunk 10]
```diff
@@ -59,54 +59,344 @@ Hazelcast Node.js client supports the following data structures and features:
...
+In order to use Hazelcast Node.js Client, we first need to setup an IMDG cluster.
+
+### Setting up an IMDG Cluster
+There are multiple ways of starting an IMDG cluster easily. You can run standalone IMDG members by downloading and running jar files
+from the website. You can embed IMDG members to your Java projects. The easiest way is to use [hazelcast-member tool](https://github.com/hazelcast/hazelcast-member-tool)
+if you have brew installed in your computer. We are going to download jars from the website and run a standalone member for this guide.
+
+#### Running Standalone Jars
+Go to https://hazelcast.org/download/ and download `.zip` or `.tar` distribution of Hazelcast IMDG. Decompress the contents into any directory that you
+want to run IMDG members from. Change into the directory that you decompressed the Hazelcast content. Go into `bin` directory. Use either
```
[the remark this answers is not in the document]
Comment (mustafaiman) [reply README.md]: actually `/bin` is not correct. It links to a bin folder in root whereas I was referring to a bin directory in current working directory. The other one is wrong, I'll change that.

**Comment by mustafaiman** [comment]
@furkansenharputlu I think I have addressed all the comments now. Can you take another look?
@Serdaro @furkansenharputlu I removed the reference to `CONFIG.md` and replaced them with references to `Configuration Overview` chapter. I guess it is better to incorporate all the information in `CONFIG.md` to here and get rid of it as you previously suggested.

**Comment by mustafaiman** [comment]
@furkansenharputlu do you have further comments?

File: README.md [hunk 11]
```diff
@@ -65,51 +65,351 @@ Hazelcast Node.js client supports the following data structures and features:
...
+An easy way to configure your Hazelcast Node.js Client is to create a `Config` object and set the appropriate options. Then you can
+supply this object to your client at the startup. Another way to configure your client is to provide a `hazelcast-client.json` file. This approach is similar to `hazelcast.xml` approach
+in configuring the member. Note that `hazelcast-client.json` is a JSON file whereas member configuration is XML based. Although these
+two formats are different, you will realize that the names of the configuration parameters are the same for both the client and member.
+It is done this way to make it easier to transfer Hazelcast skills to multiple platforms.
+
+Once you embedded `hazelcast-client` to your Node.js project, you may follow any of programmatic or declarative configuration approaches.
+We will provide both ways for each configuration option in this section. Pick one way and stick to it.
+
+---
```
Comment (furkansenharputlu) [inline README.md]: ??

**Pull request merged**